operator/src/main/java/oracle/kubernetes/operator/MakeRightDomainOperation.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator;
+
+import java.util.Optional;
+
+import oracle.kubernetes.operator.helpers.DomainPresenceInfo;
+import oracle.kubernetes.operator.work.Packet;
+import oracle.kubernetes.operator.work.Step;
+import oracle.kubernetes.weblogic.domain.model.Domain;
+
+import static oracle.kubernetes.operator.ProcessingConstants.MAKE_RIGHT_DOMAIN_OPERATION;
+
+/**
+ * Defines the operation to bring a running domain into compliance with its domain resource and introspection result.
+ */
+public interface MakeRightDomainOperation {
+
+  /**
+   * Defines the operation as pertaining to the deletion of a domain.
+   */
+  MakeRightDomainOperation forDeletion();
+
+  MakeRightDomainOperation withExplicitRecheck();
+
+  MakeRightDomainOperation withDeleting(boolean deleting);
+
+  MakeRightDomainOperation interrupt();
+
+  void execute();
+
+  Step createSteps();
+
+  void setInspectionRun();
+
+  boolean wasInspectionRun();
+
+  private static boolean wasInspectionRun(Packet packet) {
+    return fromPacket(packet).map(MakeRightDomainOperation::wasInspectionRun).orElse(false);
+  }
+
+  static void recordInspection(Packet packet) {
+    fromPacket(packet).ifPresent(MakeRightDomainOperation::setInspectionRun);
+  }
+
+  static boolean isInspectionRequired(Packet packet) {
+    return domainRequiresIntrospectionInCurrentMakeRight(packet) && !wasInspectionRun(packet);
+  }
+
+  /**
```

Comment:
checkstyle demanded that I add javadoc to this method, even though it is private. That doesn't seem right to me.